lib/__tests__/package-integrity/handler/integrity.test.ts

```diff
@@ -0,0 +1,210 @@
+import * as child from 'child_process';
+import * as os from 'os';
+import * as path from 'path';
+import AdmZip from 'adm-zip';
+import * as fs from 'fs-extra';
+import * as tar from 'tar';
+import { NpmArtifactIntegrity, PublishedPackage, PyPIArtifactIntegrity, RepositoryIntegrity } from '../../../package-integrity/handler/integrity';
+import { Repository } from '../../../package-integrity/handler/repository';
+
+function fixturePath(name: string) {
+  return path.join(__dirname, '__fixtures__', name);
+}
+
+type Download = (pkg: PublishedPackage, target: string) => void;
+
+interface IntegrityInputs {
+  npmDownload: Download;
+  pypiDownload: Download;
+  repoDir: string;
+}
+
+function initializeRepo(repoDir: string): Repository {
+
+  const shell = (command: string) => child.execSync(command, { cwd: repoDir, stdio: ['ignore', 'inherit', 'inherit'] });
+
+  // we need CI=false since projen defaults to CI=true in a GitHub
+  // environment, and project generation normally happens outside of CI.
+  const projen = () => shell('CI=false ./node_modules/.bin/projen');
+
+  const isProjen = fs.existsSync(path.join(repoDir, '.projenrc.js'));
+
+  if (isProjen) {
+    // see https://github.com/projen/projen/issues/1356
+    // eslint-disable-next-line @typescript-eslint/no-require-imports
+    const projenVersion = require(path.join(require.resolve('projen'), '..', '..', 'package.json')).version;
+    shell(`npm install --no-package-lock projen@${projenVersion}`);
```

Comment:
This extracts the version of `projen` being used by `delivlib` itself, so it can then install and use the same version when generating the test fixture projects.